From b6fd9afbf6826febd054b7038923dac74f16ad4a Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 26 Aug 2018 17:46:36 +0300
Subject: [PATCH] Add BTL package support

It is a dedicated package for BTL.
Next step is to update post-image.sh script to
use it.
---

diff --git a/Config.in b/Config.in
index 38e508f..227c5fe 100644
--- a/Config.in
+++ b/Config.in
@@ -7,6 +7,7 @@
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/dsv/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/thermal/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/preboot/Config.in"
+source "$BR2_EXTERNAL_HABANALABS_PATH/package/btl/Config.in"
 
 # Source the host tools
 source "$BR2_EXTERNAL_HABANALABS_PATH/Config.in.host"
diff --git a/board/habanalabs/common/local.mk b/board/habanalabs/common/local.mk
index 56d5244..67a272f 100644
--- a/board/habanalabs/common/local.mk
+++ b/board/habanalabs/common/local.mk
@@ -10,3 +10,4 @@
 THERMAL_OVERRIDE_SRCDIR = $(TOPDIR)/../thermal
 PREBOOT_OVERRIDE_SRCDIR = $(TOPDIR)/../preboot
 HL_FW_LOADER_OVERRIDE_SRCDIR = $(TOPDIR)/../fw_loader
+BTL_OVERRIDE_SRCDIR = $(TOPDIR)/../btl
diff --git a/configs/dali_defconfig b/configs/dali_defconfig
index 25adc1a..f7d88eb 100644
--- a/configs/dali_defconfig
+++ b/configs/dali_defconfig
@@ -65,3 +65,4 @@
 BR2_PACKAGE_THERMAL=y
 BR2_PACKAGE_PREBOOT=y
 BR2_PACKAGE_HOST_HL_FW_LOADER=y
+BR2_PACKAGE_BTL=y
diff --git a/configs/gaudi_defconfig b/configs/gaudi_defconfig
index 831893e..d5ec6fb 100644
--- a/configs/gaudi_defconfig
+++ b/configs/gaudi_defconfig
@@ -65,3 +65,4 @@
 BR2_PACKAGE_THERMAL=y
 BR2_PACKAGE_PREBOOT=y
 BR2_PACKAGE_HOST_HL_FW_LOADER=y
+BR2_PACKAGE_BTL=y
diff --git a/package/btl/Config.in b/package/btl/Config.in
new file mode 100644
index 0000000..39e92a8
--- /dev/null
+++ b/package/btl/Config.in
@@ -0,0 +1,6 @@
+config BR2_PACKAGE_BTL
+	bool "BTL"
+	help
+	  BooT Loader FW
+	  Includes Assembler and Assembly code.
+	  The Binary is executed by Boot Sequencer HW FSM.
diff --git a/package/btl/btl.mk b/package/btl/btl.mk
new file mode 100644
index 0000000..0cda194
--- /dev/null
+++ b/package/btl/btl.mk
@@ -0,0 +1,11 @@
+BTL_VERSION = 1.0.0
+BTL_SITE = btl
+BTL_LICENSE = unclear
+
+BTL_DEPENDENCIES = preboot
+
+define BTL_BUILD_CMDS
+	$(TARGET_MAKE_ENV) $(MAKE) -C $(@D) all
+endef
+
+$(eval $(generic-package))
